backend/services/application.py

```diff
@@ -0,0 +1,230 @@
+        """
+
+        if application.id:
+            application.id = None
+
+        application_entity = New_UTA_Entity.from_model(application)
+
+        application_entity.preferred_sections = (
+            self._session.query(SectionEntity)
+            .filter(
+                SectionEntity.id.in_(
+                    section.id for section in application.preferred_sections
+                )
+            )
+            .all()
+        )
+
+        self._session.add(application_entity)
+        self._session.commit()
+
+        for index, section in enumerate(application.preferred_sections):
+
+            self._session.execute(
+                update(section_application_table)
+                .filter(
+                    section_application_table.c.application_id == application_entity.id,
+                    section_application_table.c.section_id == section.id,
+                )
+                .values(preference=index)
+            )
+
+        self._session.commit()
+
+        return application_entity.to_details_model()
+
+    def update_undergrad(
+        self, subject: User, application: New_UTADetails
+    ) -> New_UTADetails:
+        """
+        Updates an application for a user based on the application sent in.
+        The application id must exist or an error is raised.
+
+        Parameters:
+            subject (User): The User that would like to update their application
+            application (New_UTADetails: The Application with the updated values
+
+        Returns:
+            New_UTADetails: Updated application
+        """
+
+        original_application = (
+            self._session.query(New_UTA_Entity)
+            .filter(ApplicationEntity.user_id == subject.id)
+            .first()
+        )
+
+        if original_application is None:
+            raise ResourceNotFoundException(f"Application does not exist")
+
+        self._session.execute(
+            delete(section_application_table).filter(
+                section_application_table.c.application_id == original_application.id
+            )
+        )
+
+        original_application.update(
+            application,
+            self._session.query(SectionEntity)
+            .filter(
+                SectionEntity.id.in_(
+                    section.id for section in application.preferred_sections
+                )
+            )
+            .all(),
+        )
+
+        self._session.commit()
+
+        for index, section in enumerate(application.preferred_sections):
+
+            self._session.execute(
+                update(section_application_table)
+                .filter(
+                    section_application_table.c.application_id
+                    == original_application.id,
+                    section_application_table.c.section_id == section.id,
+                )
+                .values(preference=index)
+            )
+        self._session.commit()
+
+        return original_application.to_details_model()
```

Comment:
Nit: Convention is to move away from `.execute()` in favor of direct session method calls, this would also make the calls in this section more readable!